redash/query_runner/query_results.py

```diff
@@ -34,9 +35,12 @@ def _load_query(user, query_id):
     if user.org_id != query.org_id:
         raise PermissionError("Query id {} not found.".format(query.id))
 
-    if not has_access(query.data_source, user, not_view_only):
-        raise PermissionError(u"You are not allowed to execute queries on {} data source (used for query id {}).".format(
-            query.data_source.name, query.id))
+    # TODO: this duplicates some of the logic we already have in the redash.handlers.query_results.
+    # We should merge it so it's consistent.
+    if not has_access(query.data_source, user, view_only):
+        raise PermissionError(
+            u"You are not allowed to execute queries on {} data source (used for query id {})."
```

Comment:
Is this error message still accurate after this change?

Reply:
https://github.com/getredash/redash/pull/4112/commits/32ba5d76790871175934db016a4f34c08d192a4e

Reply:
Oops: https://github.com/getredash/redash/pull/4112/commits/bb4f76808e206549d5397bc7e14fa852583255a9.